kratos/sources/process_info.cpp

```diff
@@ -0,0 +1,207 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Philipp Bucher
+//
+
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "includes/process_info.h"
+#include "includes/variables.h"
+
+namespace Kratos
+{
+
+using IndexType = ProcessInfo::IndexType;
+
+void ProcessInfo::SetAsTimeStepInfo()
+{
+    mIsTimeStep = true;
+    SetCurrentTime((*this)(TIME));
+}
+
+void ProcessInfo::SetAsTimeStepInfo(double NewTime)
+{
+    mIsTimeStep = true;
+    SetCurrentTime(NewTime);
+}
+
+ProcessInfo::Pointer ProcessInfo::pGetPreviousTimeStepInfo(IndexType StepsBefore)
+{
+    if(StepsBefore > 1)
+        return mpPreviousTimeStepInfo->pGetPreviousTimeStepInfo(--StepsBefore);
+
+    KRATOS_ERROR_IF(StepsBefore == 0) << "Steps before = 0" << std::endl;
+    KRATOS_ERROR_IF_NOT(mpPreviousTimeStepInfo) << "No previous time step exists." << std::endl;
```

Comment:
i don't love this way of checking if it is a nullptr. Isn't there any function to check if it exists?

Reply:
hm I don't know
I thought thats the standard way

btw this is in #2401 , where I only separate the processinfo into header/source

Reply:
ok, just dismiss my comment (although i still don't like it :-D)